### PR TITLE
feat: implement station connectivity graph based on pre-build data

### LINKED ISSUE #325: implement station connectivity graph based on pre-build data
sub task of #242

Generate new struct which implements interface of connectivitymap.Querier based on pre-build connectivity data.
/*
Package stationconnquerier provides connectivitymap.Querier interface based on pre-build connectivity data.
It needs to connect orig and destination point into station graph.

For orig/start point, based on electric vehicle's current energy level, it queries all possible reachable stations take start point as center.
For destination/end point, based on electric vehicle's max energy level, it queries all possible stations which could reach destination with maximum amount of charge.
For charge stations, it retrieves connectivity from pre-build data.
*/

### PR DESCRIPTION
**Pull request by CodeBear801**
issue: https://github.com/Telenav/osrm-backend/issues/242

# Issue

- Targeting issue: the issue will be CLOSED once the PR merged. If there is no issue that addresses the problem, please open a corresponding issue and link it here.  
Closes #325 

## Description    
Implement station connectivity graph based on pre-build data         
```
/*
Package stationconnquerier provides connectivitymap.Querier interface based on pre-build connectivity data.
It needs to connect orig and destination point into station graph.

For orig/start point, based on electric vehicle's current energy level, it queries all possible reachable stations take start point as center.
For destination/end point, based on electric vehicle's max energy level, it queries all possible stations which could reach destination with maximum amount of charge.
For charge stations, it retrieves connectivity from pre-build data.
*/
```

## Tasklist

 - [x] CHANGELOG-FORK.md entry ([CHANGELOG](https://github.com/Telenav/osrm-backend/wiki/CHANGELOG))
 - [x] [profiles/CHANGELOG.md](https://github.com/Telenav/osrm-backend/blob/master/profiles/CHANGELOG.md) entry if any `Lua` changes   
 - [x] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [x] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md)

### CODE REVIEW

File: integration/service/oasis/stationconnquerier/station_conn_querier.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	"strconv"
+
+	"github.com/Telenav/osrm-backend/integration/api/nav"
+	"github.com/Telenav/osrm-backend/integration/service/oasis/connectivitymap"
+	"github.com/Telenav/osrm-backend/integration/service/oasis/spatialindexer"
+	"github.com/Telenav/osrm-backend/integration/service/oasis/stationfinder/stationfindertype"
+	"github.com/golang/glog"
+)
+
+type StationConnectivityQuerier struct {
```
Comment (wangyoucao577) [inline integration/service/oasis/stationconnquerier/station_conn_querier.go:13]: comment missed

File: integration/service/oasis/stationconnquerier/station_conn_querier.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	stationLocationQuerier   spatialindexer.PlaceLocationQuerier
+	stationConnectivity      *connectivitymap.ConnectivityMap
+	reachableStationsByStart []*connectivitymap.QueryResult
+	reachableStationToEnd    map[string]*connectivitymap.QueryResult
+	startLocation            *nav.Location
+	endLocation              *nav.Location
+}
+
+func New(stationFinder spatialindexer.Finder, stationRanker spatialindexer.Ranker,
```
Comment (wangyoucao577) [inline integration/service/oasis/stationconnquerier/station_conn_querier.go:22]: comment

File: integration/service/oasis/stationconnquerier/station_conn_querier.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+// NearByStationQuery finds near by stations by given stationID and return them in recorded sequence
+// Returns nil if given stationID is not found or no connectivity
+func (querier *StationConnectivityQuerier) NearByStationQuery(stationID string) []*connectivitymap.QueryResult {
+
+	if stationID == stationfindertype.OrigLocationID {
+		return querier.reachableStationsByStart
+	}
+
+	if stationID == stationfindertype.DestLocationID {
+		return nil
```
Comment (wangyoucao577) [inline integration/service/oasis/stationconnquerier/station_conn_querier.go:90]: dest means not found?

**Comment by wangyoucao577** [comment]
Why implement part of #242? Can't it be implemented in one PR?     
It's also strange that the build check hasn't been triggered. Any special action you did?

File: integration/service/oasis/stationconnquerier/station_conn_querier.go (hunk 3 above)
Comment (CodeBear801) [reply integration/service/oasis/stationconnquerier/station_conn_querier.go:90]: Add more comments

```
// NearByStationQuery finds near by stations by given stationID and return them in recorded sequence
// Returns nil if given stationID is not found or no connectivity
// For start point, directly returns reachableStationsByStart which is generated by considering current energy level.
// For end point, return nil, no connectivity expected from end to others
// For charge stations, it retrieves connectivity from pre-build data.  If a charge station is reachable to destination/end point, it must connects that into graph.

```

**Comment by CodeBear801** [comment]
> Why implement part of #242? Can't it be implemented in one PR?
> It's also strange that the build check hasn't been triggered. Any special action you did?

I think its due to github.  Build it for second time solves the issue.

**Comment by wangyoucao577** [comment]
The `CHANGELOG-FORK.md entry` has been checked, but no contents?

**Pull request merged**